partner_phone_search/models/res_partner.py

```diff
@@ -0,0 +1,30 @@
+# Copyright 2018 - TODAY Serpent Consulting Services Pvt. Ltd.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl.html).
+
+from odoo import api, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+
+    @api.model
+    def name_search(self, name, args=None, operator='ilike', limit=100):
+        if not args:
+            args = []
+        if name:
+            domain = ['|', '|', ('phone', operator, name),
+                      ('mobile', operator, name), ('email', operator, name)
+                      ]
+            partners = self.search(domain + args, limit=limit,)
+            res = partners.name_get()
+            if limit:
+                limit_rest = limit - len(partners)
```

Comment:
There will be a problem if the first search returns exactly the number of partners as specified in the limit. Then limit_rest will be 0, which means an extra search will be done without any limit.
If you add the following lines after res = partners.name_get() the problem will be solved:
```
if len(partners) == limit:
    return res
```

Reply:
Setting limit_rest can be a one-liner:
`limit_rest = limit - len(partners) if limit else None`